src/placePromos.js

```diff
@@ -19,15 +19,17 @@ import { age } from './utils'
 const PROMO_PROPERTIES = ['creativeId', 'promoId', 'slotId', 'groupId', 'campaignId']
 
 /**
- * Creates a new context to run a query. The functions and objects in the
- * context will be available to the query.
+ * Extends the given context with functions and objects to be made available to
+ * the query.
  *
  * @private
  */
-function createContext (user, window) {
-  const uaParser = new UAParser(get('navigator.userAgent', window))
+function extendContext (context) {
+  const uaParser = new UAParser(get('navigator.userAgent', context.window))
```

Comment:
I think not shadowing window is a good idea :)

Reply:
I thought so too 😄